### PR TITLE
add universe finance

### PR DESCRIPTION
**Pull request by UniverseFinance**
Resolver of Universe Finance

### CODE REVIEW

File: contracts/protocols/mainnet/universeFinance/helpers.sol [hunk 1]
```diff
@@ -0,0 +1,68 @@
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+import {DSMath} from "../../common/math.sol";
```
Comment (pradyuman-verma) [inline contracts/protocols/mainnet/universeFinance/helpers.sol]:
```suggestion
import { DSMath } from "../../../utils/dsmath.sol";
```

File: contracts/protocols/mainnet/universeFinance/interface.sol [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    function getTotalAmounts() external view returns (
+        uint256 total0,
+        uint256 total1,
+        uint256 free0,
+        uint256 free1,
+        uint256 utilizationRate0,
+        uint256 utilizationRate1
+    );
+
+    function getPNL() external view override returns (uint256 rate, uint256 param);
```
Comment (pradyuman-verma) [inline contracts/protocols/mainnet/universeFinance/interface.sol]: Function has override specified but does not override anything.
```suggestion
    function getPNL() external view returns (uint256 rate, uint256 param);
```

File: contracts/protocols/mainnet/universeFinance/helpers.sol [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+import  "./interface.sol";
+
+abstract contract Helpers is DSMath {
+
+    address constant public universeReslover = 0x7466420dC366DF67b55daeDf19f8d37a346Fa7C8;
```
Comment (pradyuman-verma) [inline contracts/protocols/mainnet/universeFinance/helpers.sol]:
```suggestion
    address constant internal universeReslover = 0x7466420dC366DF67b55daeDf19f8d37a346Fa7C8;
```

File: contracts/protocols/mainnet/universeFinance/helpers.sol [hunk 4]
```diff
@@ -0,0 +1,68 @@
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+import {DSMath} from "../../common/math.sol";
+
+import  "./interface.sol";
+
+abstract contract Helpers is DSMath {
```
Comment (pradyuman-verma) [inline contracts/protocols/mainnet/universeFinance/helpers.sol]:
```suggestion
 contract Helpers is DSMath {
```

**Comment by UniverseFinance** [comment]
resolver address: 0xa7963898453c00b61cff8ce7e5b28c4e8bf1348f

**Review by KaymasJain — APPROVED** [review]

**Pull request merged**